File: dashboard/types/notification.ts
export type NotificationChannel = "email" | "webhook";
export type NotificationStatus = "pending" | "sent" | "failed" | "retrying";

export interface Notification {
  id: string;
  incident_id: string;
  channel: NotificationChannel;
  recipient: string;
  status: NotificationStatus;
  attempts: number;
  response_code: number | null;
  error_message: string | null;
  created_at: string;
  sent_at: string | null;
}

export interface SendNotificationRequest {
  incident_id: string;
  channel: NotificationChannel;
  recipient: string;
}

export interface NotificationFilters {
  status?: NotificationStatus;
  channel?: NotificationChannel;
  incident_id?: string;
  page?: number;
  per_page?: number;
}

File: mobile-app/lib/features/notifications/domain/entities/notification.dart
class Notification {
  final String id;
  final String incidentId;
  final String channel;
  final String recipient;
  final String status;
  final int attempts;
  final int? responseCode;
  final String? errorMessage;
  final DateTime createdAt;
  final DateTime? sentAt;

  const Notification({
    required this.id,
    required this.incidentId,
    required this.channel,
    required this.recipient,
    required this.status,
    required this.attempts,
    this.responseCode,
    this.errorMessage,
    required this.createdAt,
    this.sentAt,
  });

  bool get isUnread => status == 'pending';
}
